ESP8266 Beacon Client/BeaconFader.cpp

```diff
@@ -0,0 +1,59 @@
+#include "BeaconFader.h"
+#include "BeaconColor.h"
+
+BeaconFader::BeaconFader() {
+  init(BeaconColor::off(), BeaconColor::off());  
+}
+
+BeaconFader::BeaconFader(BeaconColor sourceColor, BeaconColor destinationColor) {
+  init(sourceColor, destinationColor);
+}
+
+void BeaconFader::init(BeaconColor sourceColor, BeaconColor destinationColor) {
+  this -> sourceColor = sourceColor;
+  this -> destinationColor = destinationColor;
+  red = sourceColor.getRed();
+  green = sourceColor.getGreen();
+  blue = sourceColor.getBlue();
+  white = sourceColor.getWhite();
+  rRatio = ((double)(sourceColor.getRed() - destinationColor.getRed())/((double)255)) * -1;
```

Comment:
Could just set the opposite.
```suggestion
  rRatio = -((double)(sourceColor.getRed() - destinationColor.getRed())/((double)255));
```